console

```diff
@@ -42,27 +73,103 @@ function setup {
 	fi
 
+    read -p "What is the name of the context? " CONTEXT_NAME
+    read -p "What is the IP of the cluster? " CLUSTER_IP
+    read -p "What is the username? " USERNAME
+    read -p "What is the password? " PASSWORD
+    read -p "What is the default namespace you want to use? " NAMESPACE
+
+    kubectl config set-credentials $CONTEXT_NAME-$USERNAME --username=$USERNAME --password=$PASSWORD
+    kubectl config set-cluster $CONTEXT_NAME --server=https://$CLUSTER_IP --insecure-skip-tls-verify=true
+    kubectl config set-context $CONTEXT_NAME --cluster=$CONTEXT_NAME --user=$CONTEXT_NAME-$USERNAME
+    kubectl config use-context $CONTEXT_NAME
 
+    echo "Created the context " $CONTEXT_NAME
+    echo
+    echo "Run 'kubectl cluster-info' to verify the connectivity."
+    echo
 }
 
-while [[ $# -gt 0 ]]; do
-	case $1 in
-	    -h|--help|help)
-	    help
-	    exit 0
-	    ;;
-	    setup)
-	    COMMAND=$1
-	    ;;
-	    *)
-	        echo "Unknown option: "$1
-	    ;;
-	esac
+function setup_remote {
+    REMOTE_DIR="$(dir)/.remote"
 
-	shift
-done
+    read -p "What is the namespace you are using? " NAMESPACE_NAME
+    echo $NAMESPACE_NAME > $REMOTE_DIR
+
+    read -p "What is your remote environment branch you are using? " REMOTE_BRANCH
+    echo $REMOTE_BRANCH >> $REMOTE_DIR
+
+    read -p "What is your github remote name? " REMOTE_NAME
+    echo $REMOTE_NAME >> $REMOTE_DIR
+
+    echo "Remote settings written to $REMOTE_DIR"
+}
+
+function setup {
+    setup_kubectl
+    setup_remote
+}
+
+function build {
+    ensure_new_commit
+    push_to_remote
+}
+
+function ensure_new_commit {
+    git diff --exit-code --quiet $(local_branch) $(remote_name)/$(remote_branch)
+    rc=$?; if [[ $rc = 0 ]]; then
+        echo "No changes so making timestamp update only commit to force rebuild";
+        gdate +%s > $DIR/../timestamp
+        git commit $DIR/../timestamp -m "Update timestamp to force rebuild on continuous pipe"
+    fi
+}
+
+function push_to_remote {
+    echo "Pushing to remote"
+    git push --force $(remote_name) $(local_branch):$(remote_name)
+
+    echo "Continuous Pipe will now build your developer environment"
+}
+
+function ssh {
+    cmd="kubectl --namespace=$(namespace) exec -it $(pod) -- /bin/bash"
+    echo "Running: '${cmd}'."
+    eval ${cmd}
+}
+
+function watch {
+    # Sync latency / speed in seconds
+    LATENCY="1"
+    PROJECT_DIR_WITH_DOT="$(dir)/."
+
+    # Watch for changes and sync (exclude hidden files)
+    echo    "Watching for changes. Quit anytime with Ctrl-C."
+    fswatch -0 -r -l $LATENCY .. --exclude="/\.[^/]*$" --exclude="\.idea" --exclude="\.git" --exclude="___jb_old___" --exclude="___jb_tmp___" \
+    | while read -d "" event
+      do
+        echo `date` "\"$event\" changed. Synchronizing... "
+        file="${event/$(dir)/$PROJECT_DIR_WITH_DOT}"
+        echo $result_string
+
+        rsync --relative -rlptDv -e 'kubectl --namespace='$(namespace)' exec -i '$(pod) -- $file --:/app
```

Comment:
Yeah, that probably makes sense, even tho it's not 100% sure. Actually, wouldn't it makes sense to merge the two "setup" commands so we can easily put the context in the configuration file?